### PR TITLE
LXD: Switch to GetStableRandomGenerator helper function where FNV-1a stable random numbers are generated

### PR DESCRIPTION
**Pull request by tomponline**

### CODE REVIEW

File: lxd/storage_volumes_snapshot.go [hunk 1]
```diff
@@ -733,13 +730,11 @@ func autoCreateCustomVolumeSnapshotsTask(d *Daemon) (task.Func, task.Schedule) {
...
 
-				hash := fnv.New64a()
-				_, err = io.WriteString(hash, seed)
+				r, err := util.GetStableRandomGenerator(seed)
 				if err != nil {
 					continue
 				}
 
-				r := rand.New(rand.NewSource(int64(hash.Sum64())))
 				selectedNodeID := availableNodeIDs[r.Int63n(int64(len(availableNodeIDs)))]
```
Comment (monstermunchkin) [inline lxd/storage_volumes_snapshot.go:738]: @tomponline We should use `GetStableRandomInt64FromList` here as well.

**Review by monstermunchkin — CHANGES_REQUESTED** [review]

File: lxd/storage_volumes_snapshot.go (hunk 1 above)
Comment (tomponline) [reply lxd/storage_volumes_snapshot.go]: @monstermunchkin ah yes, indeed, fixed.

File: lxd/storage_volumes_snapshot.go [hunk 2]
```diff
@@ -731,16 +728,12 @@ func autoCreateCustomVolumeSnapshotsTask(d *Daemon) (task.Func, task.Schedule) {
 
 			// If there is more than one node (clustering), a stable random node is chosen to perform the snapshot.
 			if len(nodes) > 1 {
-				seed := fmt.Sprintf("%d", v.ID)
-
-				hash := fnv.New64a()
-				_, err = io.WriteString(hash, seed)
+				nodeIDIndex, err := util.GetStableRandomInt64FromList(int(v.ID), availableNodeIDs)
```
Comment (monstermunchkin) [inline lxd/storage_volumes_snapshot.go:731]: @tomponline this is incorrect. `GetStableRandomInt64FromList` returns an actual value from the list, not just an index. It therefore should read
```go
selectedNodeID, err := util.GetStableRandomInt64FromList(int(v.ID), availableNodeIDs)
```

File: lxd/storage_volumes_snapshot.go [hunk 3]
```diff
@@ -731,16 +728,12 @@ func autoCreateCustomVolumeSnapshotsTask(d *Daemon) (task.Func, task.Schedule) {
...
 				if err != nil {
 					continue
 				}
 
-				r := rand.New(rand.NewSource(int64(hash.Sum64())))
-				selectedNodeID := availableNodeIDs[r.Int63n(int64(len(availableNodeIDs)))]
+				selectedNodeID := availableNodeIDs[nodeIDIndex]
```
Comment (monstermunchkin) [inline lxd/storage_volumes_snapshot.go:736]: @tomponline this can be removed. See comment above.

**Review by monstermunchkin — CHANGES_REQUESTED** [review]

File: lxd/storage_volumes_snapshot.go (hunk 2 above)
Comment (tomponline) [reply lxd/storage_volumes_snapshot.go]: @monstermunchkin got you thanks. fixed

**Review by monstermunchkin — APPROVED** [review]

**Pull request merged**